src/test/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApApTPCTest.java

```diff
@@ -78,41 +78,101 @@ private static DeviceDataManager createDeviceDataManager() {
 		return deviceDataManager;
 	}
 
+	/**
+	 *
+	 * Creates a data model with 3 devices in only the given band.
+	 * All are at max_tx_power, which represents the first step in greedy TPC.
+	 *
+	 * @param band band (e.g., "2G")
+	 * @return a data model
+	 */
+	private static DataModel createModelSingleBand(String band) {
+		DataModel model = new DataModel();
+
+		final int channel = UCentralUtils.LOWER_CHANNEL_LIMIT.get(band);
+
+		State stateA = TestUtils.createState(
+			channel,
+			DEFAULT_CHANNEL_WIDTH,
+			MAX_TX_POWER,
+			BSSID_A
+		);
+		State stateB = TestUtils.createState(
+			channel,
+			DEFAULT_CHANNEL_WIDTH,
+			MAX_TX_POWER,
+			BSSID_B
+		);
+		State stateC = TestUtils.createState(
+			channel,
+			DEFAULT_CHANNEL_WIDTH,
+			MAX_TX_POWER,
+			BSSID_C
+		);
+
+		model.latestState.put(DEVICE_A, stateA);
+		model.latestState.put(DEVICE_B, stateB);
+		model.latestState.put(DEVICE_C, stateC);
+
+		model.latestDeviceStatus.put(
+			DEVICE_A,
+			TestUtils
+				.createDeviceStatusSingleBand(channel, MAX_TX_POWER)
+		);
+		model.latestDeviceStatus.put(
+			DEVICE_B,
+			TestUtils
+				.createDeviceStatusSingleBand(channel, MAX_TX_POWER)
+		);
+		model.latestDeviceStatus.put(
+			DEVICE_C,
+			TestUtils
+				.createDeviceStatusSingleBand(channel, MAX_TX_POWER)
+		);
+
+		return model;
+	}
+
 	/**
 	 * Creates a data model with 3 devices. All are at max_tx_power, which
 	 * represents the first step in greedy TPC.
 	 *
 	 * @return a data model
 	 */
-	private static DataModel createModel() {
+	private static DataModel createModelDualBand() {
 		DataModel model = new DataModel();
 
+		final int channel2G =
+			UCentralUtils.LOWER_CHANNEL_LIMIT.get(UCentralConstants.BAND_2G);
+		final int channel5G =
+			UCentralUtils.LOWER_CHANNEL_LIMIT.get(UCentralConstants.BAND_5G);
+
 		State stateA = TestUtils.createState(
```

Comment:
if you like the suggestion above (purely subjective) -- you may do the same code pattern here

Reply:
I do like it, I will do it here too.